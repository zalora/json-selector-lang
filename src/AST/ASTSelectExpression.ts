// Copyright (c) ZALORA
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the root directory of this source tree.

import ASTExpression from './ASTExpression';
import { Token } from '../Lexer';

class ASTSelectExpression implements ASTExpression {
  token: Token;
  key: string;

  constructor(token: Token, key: string) {
    this.token = token;
    this.key = key;
  }

  getDescription(): string {
    return `${this.token.literal} ${this.key}`;
  }

  tokenLiteral(): string {
    const { literal = '' } = this.token || {};

    return literal;
  }
}

export default ASTSelectExpression;
