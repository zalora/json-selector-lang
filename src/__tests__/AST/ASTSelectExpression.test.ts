// Copyright (c) ZALORA
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the root directory of this source tree.

import ASTSelectExpression from '../../AST/ASTSelectExpression';
import { ident } from '../../tokens';

describe('ASTSelectExpression', () => {
  const token = { type: ident, literal: 'data' };

  it('tests instance with token input', () => {
    const astSelectExp = new ASTSelectExpression(token, token.literal);
    expect(astSelectExp).toMatchSnapshot();
  });

  it('tests toString returns value of the identifier', () => {
    const astSelectExp = new ASTSelectExpression(token, token.literal);
    expect(astSelectExp.toString()).toMatchSnapshot();
  });

  it('tests if tokenLiteral returns literal of the token', () => {
    const astSelectExp = new ASTSelectExpression(token, token.literal);
    expect(astSelectExp.tokenLiteral()).toMatchSnapshot();
  });
});
