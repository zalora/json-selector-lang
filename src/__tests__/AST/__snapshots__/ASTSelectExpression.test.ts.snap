// Copyright (c) ZALORA
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the root directory of this source tree.
//
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ASTSelectExpression tests if tokenLiteral returns literal of the token 1`] = `"data"`;

exports[`ASTSelectExpression tests instance with token input 1`] = `
ASTSelectExpression {
  "key": "data",
  "token": Object {
    "literal": "data",
    "type": Symbol(ident),
  },
}
`;

exports[`ASTSelectExpression tests toString returns value of the identifier 1`] = `"datadata"`;
